library/std/src/process.rs

```diff
@@ -1676,6 +1676,20 @@ impl ExitCode {
     pub const FAILURE: ExitCode = ExitCode(imp::ExitCode::FAILURE);
 }
 
+impl ExitCode {
+    // This should not be stabilized when stabilizing ExitCode, we don't know that i32 will serve
+    // all usecases, for example windows seems to use u32, unix uses the 8-15th bits of an i32, we
+    // likely want to isolate users anything that could restrict the platform specific
+    // representation of an ExitCode
+    //
+    // More info: https://internals.rust-lang.org/t/mini-pre-rfc-redesigning-process-exitstatus/5426
+    #[unstable(feature = "process_exitcode_placeholder", issue = "48711")]
+    /// Convert an ExitCode into an i32
+    pub fn to_i32(self) -> i32 {
```

Comment:
You probably want to mark this as `#[inline]` -- we can try perf without, but I suspect it'll mean codegen increase in `fn main() {}` style binaries as this function won't get inlined away -- currently the caller (`<ExitCode as Termination>::report`) and callee (`imp::ExitCode::as_i32`) are both marked `#[inline]`.